squad/api/rest.py

```diff
@@ -571,6 +572,29 @@ def unsubscribe(self, request, pk=None):
         data = {"email": subscriber_email}
         return Response(data, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=['get', 'post'], suffix='compare_builds')
+    def compare_builds(self, request, pk=None):
+        builds_to_compare = {f: request.GET.get(f, request.POST.get(f, None)) for f in ['baseline', 'to_compare']}
+        if all(builds_to_compare.values()):
+            try:
+                int(builds_to_compare['baseline'])
+                int(builds_to_compare['to_compare'])
+                baseline = self.get_object().builds.get(pk=builds_to_compare['baseline'])
+                to_compare = self.get_object().builds.get(pk=builds_to_compare['to_compare'])
+                if not baseline.status.finished and not to_compare.status.finished:
```

Comment:
If `baseline.status.finished == True`, the "if" will evaluate to False and you're going to be discarding the check for `to_compare.status.finished`. The operator should be "or" instead of "and".

Reply:
Good catch @chaws. I will fix this.